opm/autodiff/FlowMain.hpp

```diff
@@ -369,36 +482,75 @@ namespace Opm
             path fpath(deck_filename);
             std::string baseName;
-            std::string debugFile;
+            std::ostringstream debugFileStream;
+            std::ostringstream logFileStream;
+
             if (boost::to_upper_copy(path(fpath.extension()).string()) == ".DATA") {
                 baseName = path(fpath.stem()).string();
             } else {
                 baseName = path(fpath.filename()).string();
             }
             if (param_.has("output_dir")) {
-                logFile_ = output_dir_ + "/" + baseName + ".PRT";
-                debugFile = output_dir_ + "/." + baseName + ".DEBUG";
-            } else {
-                logFile_ = baseName + ".PRT";
-                debugFile = "." + baseName + ".DEBUG";
+                logFileStream << output_dir_ << "/";
+                debugFileStream << output_dir_ + "/";
+            }
+
+            logFileStream << baseName;
+            debugFileStream << "." << baseName;
+
+            if ( must_distribute_ && mpi_rank_ != 0 )
+            {
+                // Added rank to log file for non-zero ranks.
+                // This prevents message loss.
+                debugFileStream << "."<< mpi_rank_;
+                // If the following file appears then there is a bug.
+                logFileStream << "." << mpi_rank_;
             }
-            std::shared_ptr<EclipsePRTLog> prtLog = std::make_shared<EclipsePRTLog>(logFile_ , Log::NoDebugMessageTypes);
+            logFileStream << ".PRT";
+            debugFileStream << ".DEBUG";
+
+            std::string debugFile = debugFileStream.str();
+            logFile_ = logFileStream.str();
+
+            std::shared_ptr<EclipsePRTLog> prtLog = std::make_shared<EclipsePRTLog>(logFile_ , Log::NoDebugMessageTypes, false, output_cout_);
             std::shared_ptr<StreamLog> streamLog = std::make_shared<StreamLog>(std::cout, Log::StdoutMessageTypes);
             OpmLog::addBackend( "ECLIPSEPRTLOG" , prtLog );
             OpmLog::addBackend( "STREAMLOG", streamLog);
-            std::shared_ptr<StreamLog> debugLog = std::make_shared<EclipsePRTLog>(debugFile, Log::DefaultMessageTypes);
+            std::shared_ptr<StreamLog> debugLog = std::make_shared<EclipsePRTLog>(debugFile, Log::DefaultMessageTypes, false, output_cout_);
             OpmLog::addBackend( "DEBUGLOG" ,  debugLog);
             prtLog->setMessageFormatter(std::make_shared<SimpleMessageFormatter>(false));
             streamLog->setMessageLimiter(std::make_shared<MessageLimiter>(10));
             streamLog->setMessageFormatter(std::make_shared<SimpleMessageFormatter>(true));
             // Read parameters.
-            OpmLog::debug("\n---------------    Reading parameters     ---------------\n");
+            if ( output_cout_ )
+            {
+                OpmLog::debug("\n---------------    Reading parameters     ---------------\n");
+            }
         }
 
+        void mergeParallelLogFiles()
```

Comment:
OK here is what I would like: to have the amount of blank lines reflect how distant the things separated by the lines are. Having a single blank line between every non-blank line (as an extreme case example) makes no sense, since it gives no contrast. It is that contrast that gives an intuitive grasp of the structure of the code I want. That boils down to the following basic rules of thumb:
 - never more than a single blank line inside a function,
 - usually more than a single blank line between function definitions (and I think 4-5 is nice here).

It's a nitpick but one of my favorite ones...